cpp/include/cudf/detail/utilities/trie.cuh

```diff
@@ -89,6 +89,8 @@ inline thrust::host_vector<SerialTrieNode> createSerializedTrie(
   // Serialize the tree trie
   std::deque<IndexedTrieNode> to_visit;
   thrust::host_vector<SerialTrieNode> nodes;
+  // suport for matching empty input
+  nodes.push_back(SerialTrieNode(trie_terminating_character, tree_trie.is_end_of_word));
```

Comment:
Does this mean all tries match empty input? Difficult to follow the logic how this coordinates with the code below.

Reply:
Not always. Here we create a node that will have `is_end_of_word` = true if the root node of the `tree_trie` is also end of a word (empty word, since it's the root). Hence, passing tree_trie.is_end_of_word as the second parameter. 
I'll expand the comment above to make this clearer. :+1: